experiment/src/org/labkey/experiment/api/property/PropertyServiceImpl.java

```diff
@@ -174,6 +187,20 @@ public void registerDomainKind(DomainKind type)
         return Collections.unmodifiableList(result);
     }
 
+    @Override
+    public List<? extends Domain> getDomains(Container container, User user, Set<String> domainKinds, boolean includeProjectAndShared)
+    {
+        List<? extends Domain> domainsInContainer = getDomains(container);
+        List<Domain> result = new ArrayList<>();

```

Comment:
I prefer to avoid using forEach to build up a list in favor of a filtered stream:

```suggestion
        List<Domain> result = domainsInContainer.stream().filter(d -> d.getDomainKind() != null && domainKinds.contains(d.getDomainKind().getKindName())).collect(Collectors.toList());
```